Rewrite DnD guides (#40943)
mdn/content: files/en-us/web/api/html_drag_and_drop_api/kanban_board/index.md

## Declaring drop targets

We want to make the task columns into valid [drop targets](/en-US/docs/Web/API/HTML_Drag_and_Drop_API#drop_target) for the dragged tasks. As a baseline, we need to listen for {{domxref("HTMLElement/dragover_event", "dragover")}} and cancel it. However, we take care and only cancel the event if the drag event is dragging a task—if we are trying to drop anything else, the column should not be a drop target.

```js live-sample___kanban
const columns = document.querySelectorAll(".task-column");

columns.forEach((column) => {
  column.addEventListener("dragover", (event) => {
    // Test a custom type we will set later
    if (event.dataTransfer.types.includes("task")) {
      event.preventDefault();
    }
  });
});
```

Now, when a task is dragged over a column, you may see a [cursor effect](/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Drag_operations#drop_effects) such as a plus sign indicating that the task will be copied when dropped, because copying is the default action. Later, we will change this indicator because the task will actually be moved.

## Moving elements

> Now we implement the core functionality: the ability to move tasks between columns. It consists of two steps: add the dragged element to the target column and remove it from the source column. To do this we need to know three things when the drop happened: what the dragged element is, what the source column is, and where to insert it in the target column.

Reviewer: > what the source column is

I don't see where you need this.

Author: ```suggestion
Now we implement the core functionality: the ability to move tasks between columns. It consists of two steps: add the dragged element to the target column and remove it from the source column.
```

> We track the dragged element and the source column this way: on `dragstart`, we mark the dragged task with an `id`. Then on `drop`, we can use this ID to identify the task, and then find its source column by walking up the DOM tree. Finally we remember to remove the ID on `dragend` so we don't create duplicate IDs on a later drag.

Author: ```suggestion
We track the dragged element and the source column this way: on `dragstart`, we mark the dragged task with an `id`. Then on `drop`, we can use this ID to identify the task and remove it from the source column. Finally we remember to remove the ID on `dragend` so we don't create duplicate IDs on a later drag.
```
